### DISCUSSION TITLE
Get user id from username.

### DISCUSSION THREAD

**Discussion by tremiro** [Rapptz/discord.py#6786] [Q&A]
Does anyone know how to get someone's user id from their username.

And I want it to work on ppl outside of the serv

Thanks!

**Comment by Rapptz** [comment]
There's not enough context to answer this question. Strictly speaking this isn't possible though. You can only search members by username inside a guild not outside of it. If you're within a guild you can use [`Guild.get_member_named`](https://discordpy.readthedocs.io/en/latest/api.html#discord.Guild.get_member_named) to get a member from cache or you can use [`Guild.query_members`](https://discordpy.readthedocs.io/en/latest/api.html#discord.Guild.query_members) to fetch from the gateway. After you retrieve the member you can access their ID with the `id` attribute.

Other than that there is no other way.
